python3/agents/shared/trackers/enemy_tracker.py

```diff
@@ -41,20 +41,34 @@ def update(self, game_state):
         game_state['enemy_diameter'] = enemy_state["blast_diameter"]  # girthy boi
         game_state['enemy_is_invulnerable'] = (enemy_state['invulnerability'] - tick) > 1
 
-        # Is the enemy immediately trapped?
-        # Hard-coding immediate trap (can put in a strategy later)
-        ## Check if enemy can't move: ->check if player can place a bomb that attacks enemy: -> do it.
+
+
+        ###### Trapping Detection #####
         world = game_state['world']
         entities = game_state['entities']
-        # game_state['enemy_immediate_trapped'] = death_trap(game_state['enemy_pos'],world, entities) -> Thinks that position returns True.
         game_state['enemy_immediate_trapped'] = False
+        game_state['enemy_onestep_trapped'] = False
         world = game_state['world']
         world_width, world_height = get_world_dimension(world)
 
         # # Return any tiles that are empty around enemy
         enemy_surrounding_tiles = get_surrounding_tiles(game_state['enemy_pos'], world_width, world_height)
-        enemy_surrounding_empty_tiles = get_surrounding_empty_tiles(game_state['enemy_pos'], world, entities,
-                                                                    ignore_player=False)  # Needs to Include us
+        enemy_surrounding_empty_tiles = get_surrounding_empty_tiles(game_state['enemy_pos'], world, entities, ignore_player = False) # Needs to Include us 
+        if game_state['player_pos'] in enemy_surrounding_empty_tiles: 
```

Comment:
Ah, reason why player is inside surrounding empty tiles is because the players and enemies are actually added in the bomb tracker. Since enemy tracker is evaluated before bomb tracker, then it would return the player position because they technically haven't been added in yet. To fix this, you can just define this in the bomb tracker